### PR TITLE
docs: update migration docs for merkle path v2 changes

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Migration documentation for merkle path v2 in 08-wasm contract apis.

ref: #6496 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: docs/docs/03-light-clients/04-wasm/09-migrations.md [hunk 1]
```diff
@@ -9,6 +9,94 @@ slug: /ibc/light-clients/wasm/migrations
 
 This guide provides instructions for migrating 08-wasm versions.
 
+## From v0.2.0+ibc-go-v8.3-wasmvm-v2.0 to v0.3.0-ibc-go-v8.3-wasmvm-v2.0
```
Comment (damiannolan) [inline docs/docs/03-light-clients/04-wasm/09-migrations.md:12]: tentative, pending new branches etc

File: docs/docs/03-light-clients/04-wasm/09-migrations.md [hunk 2]
```diff
@@ -9,6 +9,94 @@ slug: /ibc/light-clients/wasm/migrations
...
+#[cw_serde]
+pub struct MerklePath {
++   #[schemars(with = "String")]
++   #[serde(with = "Base64", default)]
++   pub key_path: Vec<Bytes>,
+-   pub key_path: Vec<String>,
+}
+```
+
+## From v0.1.1+ibc-go-v7.3-wasmvm-v1.5 to v0.2.0-ibc-go-v7.6-wasmvm-v1.5
```
Comment (damiannolan) [inline docs/docs/03-light-clients/04-wasm/09-migrations.md:56]: ditto

File: docs/docs/03-light-clients/04-wasm/09-migrations.md [hunk 3]
```diff
@@ -9,6 +9,94 @@ slug: /ibc/light-clients/wasm/migrations
...
+An example of the required changes in a client contract may look like:
+
+```diff
+#[cw_serde]
+pub struct MerklePath {
++   #[schemars(with = "String")]
++   #[serde(with = "Base64", default)]
++   pub key_path: Vec<Bytes>,
+-   pub key_path: Vec<String>,
+}
```
Comment (damiannolan) [inline docs/docs/03-light-clients/04-wasm/09-migrations.md:53]: @srdtrk, could you confirm if this is correct unmarshalling syntax for `[]string` -> `[][]byte`? I can also modify the composable-ibc grandpa-cw contract and try to test this

Comment (srdtrk) [reply docs/docs/03-light-clients/04-wasm/09-migrations.md:53]: This wouldn't work

Comment (crodriguezvega) [reply docs/docs/03-light-clients/04-wasm/09-migrations.md:53]: What would be the right way to do it, @srdtrk?

**Review by crodriguezvega — APPROVED** [review]
Nice docs, pending just figuring out how to decode this on the contract side...

File: docs/docs/03-light-clients/04-wasm/09-migrations.md (hunk 3 above)
Comment (srdtrk) [reply docs/docs/03-light-clients/04-wasm/09-migrations.md:53]: We've decided to replace the usage of Bytes with `cosmwasm_std::Binary`

Comment (damiannolan) [reply docs/docs/03-light-clients/04-wasm/09-migrations.md]: https://docs.rs/cosmwasm-std/2.0.4/cosmwasm_std/struct.Binary.html

> Binary is a wrapper around Vec to add base64 de/serialization with serde. It also adds some helper methods to help encode inline.

> This is only needed as serde-json-{core,wasm} has a horrible encoding for Vec. See also https://github.com/CosmWasm/cosmwasm/blob/main/docs/MESSAGE_TYPES.md.

**Pull request merged**